### ISSUE TITLE
undo manager AssertionError

### ISSUE THREAD

**Issue by Sycophantic-Witty** [gaphor/gaphor#3348]
### Describe the bug
Selecting a relationship in the Model Browser or right-clicking a relationship in a diagram and selecting "Show in Model Browser" triggers an AssertionError.

### To Reproduce
1. Click on a relationship in the Model Browser

Time since application startup: 0:00:17
|Traceback (most recent call last):
|  File "C:\opt\dev\gaphor-2.25.1-portable\data\_internal\gaphor\transaction.pyc", line 166, in _transactional
|  File "C:\opt\dev\gaphor-2.25.1-portable\data\_internal\gaphor\transaction.pyc", line 59, in __init__
|  File "C:\opt\dev\gaphor-2.25.1-portable\data\_internal\gaphor\transaction.pyc", line 114, in _handle
|  File "C:\opt\dev\gaphor-2.25.1-portable\data\_internal\gaphor\transaction.pyc", line 190, in handle
|  File "C:\opt\dev\gaphor-2.25.1-portable\data\_internal\gaphor\application.pyc", line 191, in _transaction_proxy
|  File "C:\opt\dev\gaphor-2.25.1-portable\data\_internal\gaphor\core\eventmanager.pyc", line 77, in handle
|  File "C:\opt\dev\gaphor-2.25.1-portable\data\_internal\generic\event.pyc", line 61, in handle
|ExceptionGroup: Error while handling events (1 sub-exception)
└─┬──────────────────────────────╌┄┈
  |Traceback (most recent call last):
  |  File "C:\opt\dev\gaphor-2.25.1-portable\data\_internal\generic\event.pyc", line 57, in handle
  |  File "C:\opt\dev\gaphor-2.25.1-portable\data\_internal\gaphor\services\undomanager.pyc", line 171, in begin_transaction
  |AssertionError
  └─────────────────────────────╌┄┈

### Expected behavior
Select a relationship in either the Model Browser or a diagram and show it in the other view.

### Screenshots
N/A

### OS
- [ ] Linux (Please put in notes the specific distro)
- [ ] macOS
- [x] Windows

### Version
Version of Gaphor:

Gaphor version:         2.25.1
Operating System:       Windows (10)
Display:                GdkWin32Display
Python version:         3.12.3
GTK version:            4.14.2
Adwaita version:        1.5.0
GtkSourceView version:  5.12.0
Cairo version:          1.18.0
Pango version:          1.52.2
PyGObject version:      3.48.1
Pycairo version:        1.26.0

**Comment by amolenaar** [comment]
Hi @Sycophantic-Witty, thanks for your report.

I can reproduce it (I think).

1. Open a model containing a connected association (e.g. `test-models/all-elements.gaphor`)
2. Select the association (see where it is in the model browser)
3. Select another element
4. Click the association in the model browser
5. An error toast shows

For me, it happens as soon as I select the association. It looks like it works okay for other relationship elements it seems.

It looks like it's caused by the member end field. If I leave it empty, I see no error, if it contains a name for the association end, I get the error.

This is a new error in 2.25: before that, the property editor was not updated when you select an element in the model browser.

**Comment by Sycophantic-Witty** [comment]
Hi @amolenaar,

Today, I did not immediately faced the issue with my model (although it came back later), so I tried to reproduce it with a new model (attached).
I created two SysML blocks and a composite relationship between them in the "[bdd] Feature Domain" diagram of a new SysML model and saved with only the "Model Navigation" diagram opened.
When I restart Gaphor and open this model, the "Model Navigation" diagram appears half-updated: borders only, no text displayed.
If I now open the "[bdd] Feature Domain" diagram, I see the same kind of half-updated diagram. Clicking on the relationship between the two blocks in the "Model Browser" triggers the assertion error.
[test-model.zip](https://github.com/user-attachments/files/15835134/test-model.zip)
